EIPS/eip-5187.md

```diff
@@ -12,68 +12,85 @@ requires: 165, 1155
-     * @param expires The specified period of time to rent
+     * - `id` The id of the current token
+     * - `user` The address to rent the NFT usage rights
+     * - `amount` The amount of usage rights
+     * - `expire` The specified period of time to rent
      **/
-    event Rented(
-        uint256 indexed id,
-        address indexed user,
-        uint256 amount,
-        uint256 expires
-    );
+    event Rented(uint256 indexed id,address indexed user,uint256 amount,uint256 expire);
 
     /**
-     * @notice This emits when the NFT owner takes back the usage rights from the tenant (the `user`) 
-     * @param id The id of the current token
-     * @param user The address to rent the NFT's usage rights
-     * @param amount Amount of usage rights
+    * MUST trigger on any successful call to `renew(address user,uint256 id)`
+    *  - `id` The id of the current token
+    *  - `user` The user of the NFT
+    *  - `expire` The new specified period of time to rent
+    **/
+    event Renew(uint256 indexed id,address indexed user,uint256 expire);
+
+    /**
+    *  MUST trigger on any successful call to `renew(address user,uint256 id,uint256 expire)`
+    *  - `id` The id of the current token
+    *  - `from` The current user of the NFT
+    *  - `to` The new user
+    **/
+    event Sublet(uint256 indexed id,address indexed from,address to);
+
+    /**
+     * @notice This emits when the NFT owner takes back the usage rights from the tenant (the `user`)
+     * - id The id of the current token
+     * - user The address to rent the NFT's usage rights
+     * - amount Amount of usage rights
      **/
     event TakeBack(uint256 indexed id, address indexed user, uint256 amount);
 
     /**
      * @notice Function to rent out usage rights
-     * @param from The address to approve
-     * @param to The address to rent the NFT usage rights
-     * @param id The id of the current token
-     * @param amount The amount of usage rights
-     * @param expires The specified period of time to rent
+     * - from The address to approve
+     * - to The address to rent the NFT usage rights
+     * - id The id of the current token
+     * - amount The amount of usage rights
+     * - expire The specified period of time to rent
      **/
-    function safeRent(
-        address from,
-        address to,
-        uint256 id,
-        uint256 amount,
-        uint256 expires
-    ) external;
+    function safeRent(address from,address to,uint256 id,uint256 amount,uint256 expire) external;
 
     /**
      * @notice Function to take back usage rights after the end of the tenancy
-     * @param user The address to rent the NFT's usage rights
-     * @param tokenId The id of the current token
+     * - user The address to rent the NFT's usage rights
+     * - tokenId The id of the current token
      **/
-    function takeBack(
-        address user,
-        uint256 tokenId
-    ) external;
+    function takeBack(address user,uint256 tokenId) external;
```

Comment:
just curious, since it's a NFT, given the tokenId the contract knows who the user is. What was the reason to add a `user` as input here?

Reply:
Any user can trigger when the redemption conditions are met, and users can be notified through events.
Yes, the contract can retrieve who the user is. The design of adding 'user' here is for the convenience of the platform or other third parties to process some specific services, such as payments.